src/GovUk.Education.ExploreEducationStatistics.Common/Services/FileStoragePathUtils.cs

```diff
@@ -123,50 +104,28 @@ private static FileType GetUploadFolderForType(FileType type)
         /**
          * The admin directory path where data files are batched for importing
          */
-        public static string AdminReleaseBatchesDirectoryPath(Guid releaseId)
+        public static string AdminReleaseBatchesDirectoryPath(Guid blobPath, Guid fileId)
```

Comment:
Would suggest leaving these as `AdminRelease*` path functions as `Guid releaseId`?

Feels like these path functions should only take the specific parameters required to construct the path in question. We'd probably want to avoid letting consumers of this function worry about what a `blobPath` is supposed to be.

Reply:
I changed that one back `releaseId` and also renamed it from to `AdminReleaseBatchesDirectoryPath` `AdminDataFileBatchesDirectoryPath` as its the directory where batch files of a particular data file live.

The problem with the others is that they can all potentially be used in future to download non-release files, like image files belonging to methodology, so perhaps its their naming which is wrong, and they should be called `AdminFilePath` or something similar?

Reply:
Yeh if the intention is to make them generic enough to be used in other contexts, I'd suggest removing any mention of release from the function name :slightly_smiling_face:

Reply:
Cool I've actually been able to get rid of them all to stop it being abused so everything wanting to know the path of a file goes through extension `File.Path()`.

Tests that didn't care about the path field have had the path replaced with an arbitrary value.

Methods using `AdminDataFileBatchesDirectoryPath` have been updated to use `File.BatchesPath()` or `ReleaseFile.BatchesPath()`.

The only exceptions are

- `AdminReleaseDirectoryPath` which is renamed to a more generic `AdminFilesPath`.
- `DataArchiveValidationService` which searches for a blob existing with the future filename of the uploaded zip file, except before the database File exists. This has been updated to `var path = $"{AdminFilesPath(releaseId, DataZip)}{zipFile.FileName}";` . This validation can be removed with my next change anyway when we upload the zip by its Guid rather than filename as it can't possibly already exist.